src/Nethermind/Nethermind.Synchronization.Test/OldStyleFullSynchronizerTests.cs

```diff
@@ -61,31 +63,30 @@ public async Task Setup()
             TrieStore trieStore = new(nodeStorage, LimboLogs.Instance);
             TotalDifficultyBetterPeerStrategy bestPeerStrategy = new(LimboLogs.Instance);
             Pivot pivot = new(syncConfig);
-            BlockDownloaderFactory blockDownloaderFactory = new(
-                MainnetSpecProvider.Instance,
-                Always.Valid,
-                Always.Valid,
-                new TotalDifficultyBetterPeerStrategy(LimboLogs.Instance),
-                LimboLogs.Instance);
 
             IStateReader stateReader = new StateReader(trieStore, _codeDb, LimboLogs.Instance);
 
-            _synchronizer = new Synchronizer(
-                dbProvider,
-                nodeStorage,
-                MainnetSpecProvider.Instance,
-                _blockTree,
-                _receiptStorage,
-                _pool,
-                stats,
-                syncConfig,
-                blockDownloaderFactory,
-                pivot,
-                Substitute.For<IProcessExitSource>(),
-                bestPeerStrategy,
-                new ChainSpec(),
-                stateReader,
-                LimboLogs.Instance);
+            IServiceCollection serviceCollection = new ServiceCollection()
+                .AddSingleton(nodeStorage)
+                .AddSingleton<ISpecProvider>(MainnetSpecProvider.Instance)
+                .AddSingleton(_blockTree)
+                .AddSingleton(_receiptStorage)
+                .AddSingleton(_pool)
+                .AddSingleton<INodeStatsManager>(stats)
+                .AddSingleton<ISyncConfig>(syncConfig)
+                .AddSingleton<IBlockValidator>(Always.Valid)
+                .AddSingleton<ISealValidator>(Always.Valid)
+                .AddSingleton<IPivot>(pivot)
+                .AddSingleton(Substitute.For<IProcessExitSource>())
+                .AddSingleton<IBetterPeerStrategy>(bestPeerStrategy)
+                .AddSingleton(new ChainSpec())
+                .AddSingleton(stateReader)
+                .AddSingleton<IBeaconSyncStrategy>(No.BeaconSync)
+                .AddSingleton<ILogManager>(LimboLogs.Instance);
+            dbProvider.ConfigureServiceCollection(serviceCollection);
+
+            _synchronizer = new Synchronizer(serviceCollection, syncConfig);
```

Comment:
Test code shows that now `Synchronizer` is completely opaque: there is no way to know that we have to build a `IServiceCollection` with those specific services without looking into the implementation.